packages/trader/src/Modules/Trading/Containers/trade.jsx

```diff
@@ -159,86 +163,103 @@ const Trade = observer(() => {
-                            className={classNames({ 'vanilla-trade-chart': is_vanilla })}
-                        >
-                            {show_digits_stats && <DigitsWidget digits={digits} tick={tick} />}
-                            <ChartTrade
-                                topWidgets={topWidgets}
-                                charts_ref={charts_ref}
-                                bottomWidgets={show_digits_stats ? bottomWidgets : undefined}
-                                is_accumulator={is_accumulator}
-                            />
-                        </SwipeableWrapper>
-                    </MobileWrapper>
-                </React.Suspense>
-
-                {/* Remove Test component for debugging below for production release */}
-                <Test />
-            </Div100vhContainer>
-            <div className={form_wrapper_class}>
-                {is_market_closed && !is_market_unavailable_visible && (
-                    <MarketIsClosedOverlay
-                        is_eu={is_eu}
-                        is_synthetics_trading_market_available={is_synthetics_trading_market_available}
-                        {...(is_eu && category)}
-                        onClick={onTryOtherMarkets}
-                        onMarketOpen={prepareTradeStore}
-                        symbol={symbol}
+                    <NotificationMessages />
+                    <React.Suspense
+                        fallback={<ChartLoader is_dark={is_dark_theme} is_visible={!symbol || is_chart_loading} />}
+                    >
+                        <DesktopWrapper>
+                            <div
+                                className={classNames('chart-container__wrapper', {
+                                    'vanilla-trade-chart': is_vanilla,
+                                })}
+                            >
+                                <ChartLoader is_visible={is_chart_loading || should_show_active_symbols_loading} />
+                                <ChartTrade
+                                    topWidgets={topWidgets}
+                                    charts_ref={charts_ref}
+                                    is_accumulator={is_accumulator}
+                                />
+                            </div>
+                        </DesktopWrapper>
+                        {!(openLaunchModal && is_logged_in && !is_already_shown) && (
+                            <MobileWrapper>
+                                <ChartLoader is_visible={is_chart_loading || should_show_active_symbols_loading} />
+                                <SwipeableWrapper
+                                    onChange={onChangeSwipeableIndex}
+                                    is_disabled={
+                                        !show_digits_stats ||
+                                        !is_trade_enabled ||
+                                        form_components.length === 0 ||
```

Comment:
I know, it was there before, but can be improved a bit:
`!form_components.length`
Completely optional, can be ignored)